### DISCUSSION TITLE
Hololens 2: How to detect Hololens' height?

### DISCUSSION THREAD

**Discussion by A-Ivan** [microsoft/MixedRealityToolkit-Unity#10422] [Q&A] [answered]
Hi,

Are there any suggestions on how to detect the Hololens 2's distance from the floor?

The [Spatial awareness](https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/features/spatial-awareness/spatial-awareness-getting-started?view=mrtkunity-2021-05)/[mapping](https://docs.microsoft.com/en-us/windows/mixed-reality/develop/unity/spatial-mapping-in-unity?tabs=xr) solutions seem to be a bit of an overkill, slow, and resource heavy, as they either scan the environment and reconstruct it in 3D, which you probably then need to then aim and raycast at the ground and extract the height from that distance, or have the mapping system label your real-world environment and then query for where the ground is.

There used to be simpler solution to detect the ground for the [Hololens 1](https://localjoost.github.io/finding-floor-and-displaying-holograms/), but the functions required for this to work in the Mixed Reality Toolkit are now deprecated.
 

I know this is a third-party, but using Vuforia's [Ground Plane Detection](https://library.vuforia.com/articles/Solution/ground-plane-guide.html) seemed like a good solution, but it is currently not supported on the Hololens 2.

Aside from starting the Hololens on the floor and then putting it on (to detect height change) or using Hololens' [QR Code tracking feature](https://docs.microsoft.com/en-us/windows/mixed-reality/develop/advanced-concepts/qr-code-tracking-overview) to track a QR code placed on the floor and using that to detect its distance from the Hololens, are there any suggestions on how to detect the Hololens' current height from the ground?

Thank you,
Ivan

**Comment by polar-kev** [comment, marked as answer]
Hey Ivan, unfortunately I don't think there are any better solutions than what you've suggested. I'll bring this up for discussion internally.

**Reply by A-Ivan** [reply]
Hi @polar-kev , I understand. I would have liked for a simpler way, at least from the developing standpoint, to be able to call a function to detect the Hololens height and/or detect where the floor is (which would then be used to calculate floor's height difference to the Hololens).
